File: gateway/src/main/java/com/example/gateway/controller/UserController.java
package com.example.gateway.controller;

import com.example.api.UserApi;
import com.example.api.dto.UserDto;
import com.example.gateway.client.UserClient;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping(path = UserApi.PATH)
@RequiredArgsConstructor
public class UserController implements UserApi {

    private final UserClient userClient;

    @Override
    public UserDto create(UserDto userDto) {
        return userClient.create(userDto);
    }

    @Override
    public UserDto update(Long id, UserDto userDto) {
        return userClient.update(id, userDto);
    }

    @Override
    public UserDto findById(Long id) {
        return userClient.findById(id);
    }

    @Override
    public List<UserDto> findAll() {
        return userClient.findAll();
    }

    @Override
    public Map<String, String> deleteById(Long id) {
        return userClient.deleteById(id);
    }
}

File: gateway/src/main/kotlin/com/example/gateway/controller/UserController.kt
package com.example.gateway.controller

import com.example.api.UserApi
import com.example.api.model.UserDto
import com.example.gateway.client.UserClient
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = [UserApi.PATH])
class UserController(
    val userClient: UserClient
) : UserApi {

    override fun create(userDto: UserDto): UserDto {
        return userClient.create(userDto)
    }

    override fun update(id: Long, userDto: UserDto): UserDto {
        return userClient.update(id, userDto)
    }

    override fun findById(id: Long): UserDto {
        return userClient.findById(id)
    }

    override fun findAll(): List<UserDto> {
        return userClient.findAll()
    }

    override fun deleteById(id: Long) {
        userClient.deleteById(id)
    }
}
